plutus-core/cost-model/budgeting-bench/Benchmarks/Arrays.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE BlockArguments   #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Benchmarks.Arrays (makeBenchmarks) where
+
+import Prelude
+
+import Common
+import Control.Monad (replicateM)
+import Criterion.Main (Benchmark)
+import Data.Vector.Strict (Vector)
+import Data.Vector.Strict qualified as Vector
+import PlutusCore.Builtin (mkTyBuiltin)
+import PlutusCore.Core (Type)
+import PlutusCore.Default (DefaultFun (IndexArray, LengthOfArray, ListToArray), DefaultUni)
+import PlutusCore.Evaluation.Machine.ExMemoryUsage (ArrayCostedByLength (..),
+                                                    ListCostedByLength (..))
+import PlutusCore.Name.Unique (TyName)
+import System.Random.Stateful (StdGen, UniformRange (uniformRM), runStateGen_)
+
+--------------------------------------------------------------------------------
+-- Benchmarks ------------------------------------------------------------------
+
+makeBenchmarks :: StdGen -> [Benchmark]
+makeBenchmarks gen =
+  [ benchLengthOfArray gen
+  , benchListToArray gen
+  , benchIndexArray gen
+  ]
+
+benchLengthOfArray :: StdGen -> Benchmark
+benchLengthOfArray gen =
+  createOneTermBuiltinBench LengthOfArray [tyArrayOfInteger] listOfArrays
```

Comment:
This should use the same wrappper as the denotation.  It won't make any difference because it's constant time, but let's keep things consistent anyway.
```suggestion
  createOneTermBuiltinBenchWithWrapper
    ArrayCostedByLength
    LengthOfArray 
    [tyArrayOfInteger] 
    listOfArrays
```